src/main/java/com/appdynamics/extensions/webspheremq/metricscollector/InquireChannelCmdCollector.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.appdynamics.extensions.webspheremq.metricscollector;
+
+import com.appdynamics.extensions.MetricWriteHelper;
+import com.appdynamics.extensions.conf.MonitorContextConfiguration;
+import com.appdynamics.extensions.logging.ExtensionsLoggerFactory;
+import com.appdynamics.extensions.metrics.Metric;
+import com.appdynamics.extensions.webspheremq.config.ExcludeFilters;
+import com.appdynamics.extensions.webspheremq.config.QueueManager;
+import com.appdynamics.extensions.webspheremq.config.WMQMetricOverride;
+import com.google.common.collect.Lists;
+import com.ibm.mq.constants.CMQCFC;
+import com.ibm.mq.constants.MQConstants;
+import com.ibm.mq.headers.pcf.MQCFIL;
+import com.ibm.mq.headers.pcf.PCFException;
+import com.ibm.mq.headers.pcf.PCFMessage;
+import com.ibm.mq.headers.pcf.PCFMessageAgent;
+import com.singularity.ee.agent.systemagent.api.exception.TaskExecutionException;
+import org.slf4j.Logger;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This class is responsible for channel inquiry metric collection.
+ */
+public class InquireChannelCmdCollector extends MetricsCollector {
+
+	public static final Logger logger = ExtensionsLoggerFactory.getLogger(InquireChannelCmdCollector.class);
+	private static final String ARTIFACT = "Channels";
+
+	public InquireChannelCmdCollector(Map<String, WMQMetricOverride> metricsToReport, MonitorContextConfiguration monitorContextConfig, PCFMessageAgent agent, QueueManager queueManager, MetricWriteHelper metricWriteHelper) {
+		super(metricsToReport, monitorContextConfig, agent, metricWriteHelper, queueManager, null, ARTIFACT);
+	}
+
+	public void run() {
+		try {
+			logger.info("Collecting metrics for command MQCMD_INQUIRE_CHANNEL");
+			this.process();
+		} catch (TaskExecutionException e) {
+			logger.error("Error in ChannelMetricsCollector ", e);
+		} finally {
+			countDownLatch.countDown();
+		}
+	}
```

Comment:
The `logger.error()` line is reporting the wrong class name -- but whatever, I've been removing these `run()` blocks in favor of this new `MetricsPublisherJob` class. Any usages that need to "run" this collector should pass an instance of this to an instance of a `MetricsPublisherJob` instead.

I suppose it doesn't matter in this PR...but cleanup is happening in this package and it'll get swept up eventually.

Reply:
Oh I think you're already doing that, so you can safely delete this run() method. 🤘🏻

Reply:
Yes, I thought I did. Weird.